indra/newview/llstartup.cpp

```diff
@@ -400,10 +401,12 @@ bool idle_startup()
         static bool first_call = true;
         if (first_call)
         {
+            first_call = false;
             // Other phases get handled when startup state changes,
             // need to capture the initial state as well.
             LLStartUp::getPhases().startPhase(LLStartUp::getStartupStateString());
-            first_call = false;
+            // Use our custom scheduler for coroutine scheduling.
+            llcoro::scheduler::use();
```

Comment:
Maybe move the call somewhere here: https://github.com/secondlife/viewer/blob/ec092bb626ecb2f293343e35209b31e363de6a5d/indra/newview/llappviewer.cpp#L768

Reply:
Thanks, I like that suggestion.

Reply:
Well, I think I understood scheduling enough to say that it looks good.

On the side note.. With these changes, it will be interesting to disable our 'infinite-loop-protection' and try something like that in Lua script:
```
while true do
 print("1")
end
```

Reply:
Thank you!

I definitely value the interrupt feature that ensures other viewer coroutines get a chance even when a Lua C++ coroutine is compute-bound. I think that integrates nicely with this feature.

I still think it's a good idea to kill off a Lua coroutine that's compute-bound for too long, but maybe we can raise the limit?

Reply:
I suggest leaving the current limit as it is. We can raise it if we ever encounter a real 'non-synthetic' script that exceeds that limit.

Reply:
Heh - I've hit that already. You can see in [test_callables.lua](https://github.com/secondlife/viewer/blob/release/luau-scripting/indra/newview/scripts/lua/test_callables.lua) that the script explicitly iterates through the table returned by `UI.callables()`. That's because the simpler construct
```
print(inspect(UI.callables()))
```
exceeded the timeout during `inspect()` processing.

Reply:
Ok, now I completely agree with raising the limit.